robot/web/web.go

```diff
@@ -365,42 +364,44 @@ func (svc *webService) updateResources(resources map[resource.Name]resource.Reso
 		groupedResources[n.API] = r
 	}
 
-	apiRegs := resource.RegisteredAPIs()
-	for a, v := range groupedResources {
-		apiResColl, ok := svc.services[a]
-		// TODO(RSDK-144): register new service if it doesn't currently exist
+	// All known APIs have pre-registered resource collections, so loop through them. We want to empty out any collections if
+	// there are no longer resources available in that API.
+	for api, coll := range svc.services {
+		group, ok := groupedResources[api]
```

Comment:
Further suggested documentation slowing things down even more (for dull readers like myself). Which intentionally lay out the possibilities/consequences of this lookup:
```
For a given API that the web service has resources for, we get the new set of resources we should be updated with. If we find a set of resources, `coll.ReplaceAll` will do the work of adding any new resources and deleting old ones.

If there are no input resources of the given API, we call `coll.ReplaceAll` with an empty input such that it will remove any existing resources.
```

Reply:
cool, replaced the comment